orchagent/bufferorch.cpp

```diff
@@ -520,17 +551,292 @@ task_process_status BufferOrch::processEgressBufferProfileList(Consumer &consume
 }
 
+/*
+Input sample "BUFFER_PORT_CABLE_LENGTH_TABLE:"
+             "port_alias" : "cable_length"
+*/
+task_process_status BufferOrch::processPortCableLenth(Consumer &consumer)
+{
+    SWSS_LOG_ENTER();
+
+    KeyOpFieldsValuesTuple tuple = consumer.m_toSync.begin()->second;
+    string key = kfvKey(tuple);
+    string op = kfvOp(tuple);
+
+    if (op != SET_COMMAND)
+    {
+        return task_process_status::task_invalid_operation;
+    }
+
+    for (auto itp : kfvFieldsValues(tuple))
+    {
+        Port port;
+        string port_name = fvField(itp);
+        string cable_length = fvValue(itp);
+
+        if (!m_portsOrch->getPort(port_name, port))
+        {
+            SWSS_LOG_ERROR("Port with alias:%s not found", port_name.c_str());
+            return task_process_status::task_invalid_entry;
+        }
+
+        m_cableLengthMap[port_name] = cable_length;
+    }
+
+    return task_process_status::task_success;
+}
+
+/*
+Input sample "PORT_CONFIG_TO_PG_PROFILE_TABLE:"
+             "40000_5"          : "[BUFFER_PROFILE_TABLE:pg_lossless_40G_5m_profile]",
+             "<speed>_<length>" : "[profile_name]",
+             ...
+*/
+task_process_status BufferOrch::processPortConfigToPgProfile(Consumer &consumer)
+{
+    SWSS_LOG_ENTER();
+
+    KeyOpFieldsValuesTuple tuple = consumer.m_toSync.begin()->second;
+    string table_name = consumer.m_consumer->getTableName();
+    //sai_object_id_t buffer_profile_id;
+    string op = kfvOp(tuple);
+
+    SWSS_LOG_DEBUG("Processing:%s", table_name.c_str());
+
+    if (op != SET_COMMAND)
+    {
+        return task_process_status::task_invalid_operation;
+    }
+
+    for (auto it : kfvFieldsValues(tuple))
+    {
+        string port_config = fvField(it);
+        string buffer_profile_name = fvValue(it);
+
+        m_portConfigProfileMap[port_config] = buffer_profile_name;
+    }
+
+    return task_process_status::task_success;
+}
+
+/*
+Read string value fron the DB for given table/key
+*/
+bool BufferOrch::getTableValue(string table_name, string table_key, string item_key, string &item_value)
+{
+    SWSS_LOG_ENTER();
+
+    vector<FieldValueTuple> values;
+    swss::Table table(&m_db, table_name);
+
+    table.get(table_key, values);
+
+    for(auto it : values)
+    {
+        if (fvField(it) == item_key)
+        {
+            item_value = fvValue(it);
+            return true;
+        }
+    }
+
+    return false;
+}
+
+/*
+Read integer value fron the DB for given table/key
+*/
+bool BufferOrch::getTableValue(string table_name, string table_key, string item_key, uint32_t &item_value)
+{
+    SWSS_LOG_ENTER();
+
+    string str_item_value;
+
+    bool status = getTableValue(table_name, table_key, item_key, str_item_value);
+    if (status)
+    {
+        item_value = stoul(str_item_value);
+    }
+
+    return status;
+}
+
+/*
+Write string value fron the DB for given table/key
+*/
+bool BufferOrch::setTableValue(string table_name, string table_key, string item_key, string &item_value)
```

Comment:
this function is very generic. it is better to move it to common library instead of putting it here.